tests/queries/models.py

```diff
@@ -737,3 +737,57 @@ class RelatedIndividual(models.Model):
 
     class Meta:
         db_table = 'RelatedIndividual'
+
+
+class SchemaQualified(models.ModelTable):
+    def __init__(self, schema, table):
+        self.schema = schema
+        self.table = table
+
+    def as_sql(self, compiler, connection):
+        # Note that it would be easy to change the schema using thread locals
+        # if wanted.
+        return '%s.%s' % (connection.ops.quote_name(self.schema), connection.ops.quote_name(self.table)), []
+
+
+class SchemaQualified(models.Model):
+    val = models.TextField()
+
+    class Meta:
+        table_cls = SchemaQualified('other_schema', 'schema_qualified')
+        managed = False
+
+
+class DynamicQuery(models.ModelTable):
+    def as_sql(self, compiler, connection):
+        db_time = compiler.query.context.get('db_time')
```

Comment:
So I'm not sure I understand where context comes from: sure, it's an attribute on sql.Query, but how do we add things to the context?

I was only able to find a couple of references to it in the source, and they weren't especially instructive.